eos-updater/apply.c

```diff
@@ -96,6 +96,85 @@ get_test_osname (void)
   return g_getenv ("EOS_UPDATER_TEST_UPDATER_OSTREE_OSNAME");
 }
 
+static gboolean
+update_remote_collection_id (OstreeRepo   *repo,
+                             const gchar  *refspec,
+                             GError      **error)
+{
+  g_autofree gchar *remote = NULL;
+  g_autofree gchar *ref = NULL;
+  g_autofree gchar *cur_collection_id = NULL;
+  g_autofree gchar *checksum = NULL;
+  g_autofree gchar *remote_group = NULL;
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) metadata = NULL;
+  g_autoptr(GVariant) collection_id_v = NULL;
+  g_autoptr(GKeyFile) config = NULL;
+  const char *collection_id;
+
+  if (!ostree_parse_refspec (refspec, &remote, &ref, error))
+    return FALSE;
+  if (remote == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Invalid refspec ‘%s’ in origin: did not contain a remote name",
+                   refspec);
+      return FALSE;
+    }
+
+  /* If the remote already has a collection ID set, there's nothing to do */
+  if (!ostree_repo_get_remote_option (repo, remote, "collection-id", NULL,
+                                      &cur_collection_id, error))
+    return FALSE;
+  if (cur_collection_id != NULL && *cur_collection_id != '\0')
+    return TRUE;
```

Comment:
If someone switches between repo stages, then this might not be accurate anymore.

Reply:
When you switch changes, `eos-stage-ostree` changes the collection ID for you. I don't think that's eos-updater's job

Reply:
Agreed, that should definitely be fixed by `eos-stage-ostree`. I really don’t want any code in the system overwriting collection IDs when they are non-empty or set, as the collection ID is the root of a lot of trust. Anything automated should only ever be able to set a collection ID when it was previously unset, and should not be able to change or remove set collection IDs.